src/App.jsx

```diff
@@ -1,13 +1,58 @@
+import { useState } from 'react';
+import {
+	Switch,
+	Route,
+	BrowserRouter as Router,
+	Redirect,
+} from 'react-router-dom';
+
 import Header from './components/Header/Header';
+import Login from './components/Login/Login';
+import Registration from './components/Registration/Registration';
+import CourseInfo from './components/CourseInfo/CourseInfo';
+import CreateCourse from './components/CreateCourse/CreateCourse';
 import Courses from './components/Courses/Courses';
+import { AppContext } from './AppContext';
 
 import './App.css';
+import AuthenticatedRoute from './components/Routes/AuthenticatedRoute';
 
 function App() {
+	const [isAuthenticated, setIsAuthenticated] = useState(false);
```

Comment:
Please check this out: I logged in, got to Courses page, I refresh the page, my token is in local storage, so I want to see Course page and not to be redirected to Login form. So I suppose isAuthenticated and setIsAuthenticated are redundant at all, isAuthenticated === having token in localStorage but not setting it manually

Reply:
As for the isAuthenticated and setIsAuthenticated i find it a bit more tricky. When I use the localStorage, it's harder to remove the logout button when user logs out. I dunno...

Reply:
you can remove it at all, it is not necessary, what I mean anywhere where you use isAuthenticated you can create local varibale const isAuthenticated  = !!localStorage.getItem('authKey')

Reply:
Sorry I've just checked again, yes, you were right, revert const [isAuthenticated, setIsAuthenticated] to previous false, just check that once you logged in user name is not displayed. it appears only after refresh. Anyway keep in mind that after login and getting to the course page the user should refresh the page and stay there (with username displayed on the scrren)

Reply:
Right. Changed it and now it shows userName without need to refresh page